app/src/main/java/nerd/tuxmobil/fahrplan/congress/schedule/LectureViewDrawer.kt

```diff
@@ -0,0 +1,232 @@
+            colorResId ?: R.color.event_border_highlight
+        } else {
+            val colorResId = trackNameBackgroundColorDefaultPairs[event.track]
+            colorResId ?: R.color.event_border_default
+        }
+        @ColorInt val backgroundColor = ContextCompat.getColor(context, backgroundColorResId)
+        val eventDrawable: EventDrawable
+        eventDrawable = if (eventIsFavored && alternativeHighlightingIsEnabled) {
+            EventDrawable(
+                    backgroundColor,
+                    eventDrawableCornerRadius.toFloat(),
+                    eventDrawableRippleColor,
+                    eventDrawableStrokeColor,
+                    eventDrawableStrokeWidth.toFloat())
+        } else {
+            EventDrawable(
+                    backgroundColor,
+                    eventDrawableCornerRadius.toFloat(),
+                    eventDrawableRippleColor)
+        }
+        eventDrawable.setLayerInset(EventDrawable.BACKGROUND_LAYER_INDEX,
+                eventDrawableInsetLeft,
+                eventDrawableInsetTop,
+                eventDrawableInsetRight,
+                0)
+        eventDrawable.setLayerInset(EventDrawable.STROKE_LAYER_INDEX,
+                eventDrawableInsetLeft,
+                eventDrawableInsetTop,
+                eventDrawableInsetRight,
+                0)
+        eventView.setBackgroundDrawable(eventDrawable)
+        val padding = eventPadding
+        eventView.setPadding(padding, padding, padding, padding)
+    }
+
+    companion object {
+        const val LOG_TAG = "LectureViewDrawer"
+        const val DIVISOR = 5
+        private const val MILLIS_PER_MINUTE = 60000
+
+        @JvmStatic
+        fun setLectureTextColor(lecture: Lecture, view: View) {
+            val title = view.findViewById<TextView>(R.id.event_title)
+            val subtitle = view.findViewById<TextView>(R.id.event_subtitle)
+            val speakers = view.findViewById<TextView>(R.id.event_speakers)
+            val colorResId = if (lecture.highlight) R.color.event_title_highlight else R.color.event_title
+            val textColor = ContextCompat.getColor(view.context, colorResId)
+            title.setTextColor(textColor)
+            subtitle.setTextColor(textColor)
+            speakers.setTextColor(textColor)
+        }
+
+        @JvmStatic
+        fun calculateLayoutParams(roomIndex: Int, lectures: List<Lecture>, standardHeight: Int, conference: Conference): Map<Lecture, LinearLayout.LayoutParams> {
+            var endTimePreviousLecture: Int = conference.firstEventStartsAt
+            var startTime: Int
+            var margin: Int
+            var previousLecture: Lecture? = null
+            val lectureLayoutParams = mutableMapOf<Lecture, LinearLayout.LayoutParams>()
+
+            for (idx in lectures.indices) {
+                val lecture = lectures[idx]
+
+                if (lecture.roomIndex == roomIndex) {
```

Comment:
These lines can be rewritten to:

``` kotlin
lectures.filter { it.roomIndex == roomIndex }
        .forEachIndexed { idx, lecture ->
```

Reply:
Actually it can't. For correct margin calculation, we have to loop through all lectures.

Reply:
@johnjohndoe I had another look at it. Your approach could technically work. But, the thing is that previousLecture would be outside the .forEachIndexed closure, which is not too bad, but still an obstacle. Besides that, we would loop almost twice as much the whole lectures collection (once for each room + once for each filtered lecture).

Reply:
Yes. I stumbled into `previousLecture`, too. When looping is a real performance issue we could think about using [`Sequences`](https://kotlinlang.org/docs/reference/sequences.html).

Reply:
The issue here is, that it does not really matter how the loop is written at all. Both variations are easy to grasp. I won't make syntactic sugar changes that come with the cost of such a "performance" penalty.